src/config/env.ts

```diff
@@ -1,34 +1,75 @@
-import dotenv from 'dotenv';
-import path from 'path';
-import envVars from './env';
+import dotenv from "dotenv";
+import path from "path";
+import { LogLevelString } from "bunyan";
 
-const filename = process.env.NODE_ENV === 'test' ? '.env.test' : '.env';
-const env = dotenv.config({
-  path: path.resolve(process.cwd(), filename),
-});
+export enum EnvironmentEnum {
+  test = "test",
+  development = "development",
+  production = "production",
+}
 
-function getProxyFromEnvironment(): string{
-  let proxyHost: string = process.env.EXTERNAL_ONLY_PROXY_HOST;
-  let proxyPort: string = process.env.EXTERNAL_ONLY_PROXY_PORT;
+export enum BooleanEnum {
+  true = "true",
+  false = "false",
+}
 
-  if(!proxyHost || !proxyPort){
-    return '';
-  }
-  const proxyAddress = `http://${proxyHost}:${proxyPort}`;
+const nodeEnv: EnvironmentEnum = EnvironmentEnum[process.env.NODE_ENV];
 
-  return proxyAddress;
+const filename = nodeEnv === EnvironmentEnum.test ? ".env.test" : ".env";
+const env = dotenv.config({
+  path: path.resolve(process.cwd(), filename)
+});
+
+// TODO: add checks for environment variables here and error out if missing any
+if (env.error && nodeEnv !== EnvironmentEnum.production) {
+  throw env.error;
 }
 
-export default {
-  MICROS_ENV: process.env.MICROS_ENV || 'development',
+const getProxyFromEnvironment = (): string => {
+  const proxyHost = process.env.EXTERNAL_ONLY_PROXY_HOST;
+  const proxyPort = process.env.EXTERNAL_ONLY_PROXY_PORT;
+  return proxyHost && proxyPort ? `http://${proxyHost}:${proxyPort}` : undefined;
+};
+
+// TODO: Make envvars dynamic
+const envVars: EnvVars = {
+  MICROS_ENV: EnvironmentEnum[process.env.MICROS_ENV || EnvironmentEnum.development],
   MICROS_SERVICE_VERSION: process.env.MICROS_SERVICE_VERSION,
-  NODE_ENV: process.env.NODE_ENV,
+  NODE_ENV: nodeEnv,
   SENTRY_DSN: process.env.SENTRY_DSN,
   PROXY: getProxyFromEnvironment(),
-  ...env.parsed,
-};
+  MAINTENANCE_MODE: BooleanEnum[process.env.MAINTENANCE_MODE],
```

Comment:
I believe we should control this with a feature flag rather than env variable. Not sure if we integrated our service with Launch Darkly already, but we should definitely look into it for this case.

(However if we are not integrated, we can do it in the next PR, hence this one is quite big already)

Reply:
We don't have feature flags enabled yet.  Environment variables should be easy enough to change though for the time being.